sdk/storage/storage-blob/src/credentials/RawTokenCredential.ts

```diff
@@ -26,39 +23,36 @@ import { TokenCredentialPolicy } from "../policies/TokenCredentialPolicy";
  *    }
  *  }, 60 * 60 * 1000); // Set an interval time before your token expired
  * @export
- * @class TokenCredential
- * @extends {Credential}
+ * @implements {TokenCredential}
  *
  */
-export class TokenCredential extends Credential {
+export class RawTokenCredential implements TokenCredential {
   /**
    * Mutable token value. You can set a renewed token value to this property,
    * for example, when an OAuth token is expired.
    *
    * @type {string}
-   * @memberof TokenCredential
    */
   public token: string;
 
   /**
    * Creates an instance of TokenCredential.
    * @param {string} token
-   * @memberof TokenCredential
    */
   constructor(token: string) {
-    super();
     this.token = token;
   }
 
   /**
-   * Creates a TokenCredentialPolicy object.
-   *
-   * @param {RequestPolicy} nextPolicy
-   * @param {RequestPolicyOptions} options
-   * @returns {TokenCredentialPolicy}
-   * @memberof TokenCredential
+   * Retrieves the token stored in this RawTokenCredential.
+   * 
+   * @param _scopes Ignored since token is already known.
+   * @param _options Ignored since token is already known.
    */
-  public create(nextPolicy: RequestPolicy, options: RequestPolicyOptions): TokenCredentialPolicy {
-    return new TokenCredentialPolicy(nextPolicy, options, this);
+  getToken(_scopes: string | string[], _options?: GetTokenOptions): Promise<AccessToken | null> {
```

Comment:
>getToken [](start = 2, length = 8)

Should add public and async markers for this method `public async getToken()`

Reply:
Thanks!  Forgot the add the `public` marker. `async` shouldn't be necessary in this case since no `await` is used.

Reply:
This method returns a Promise, so I assume it should be an async method. With async marker, we can skip `Promise.resolve` wrapper.

Reply:
OK, I will change it to `async` and return the object directly.